src/test/ui/consts/std/cell.rs

```diff
@@ -1,18 +1,26 @@
+#![feature(const_refs_to_cell)]
+
 use std::cell::*;
 
 // not ok, because this would create a silent constant with interior mutability.
 // the rules could be relaxed in the future
 static FOO: Wrap<*mut u32> = Wrap(Cell::new(42).as_ptr());
-//~^ ERROR cannot borrow a constant which may contain interior mutability
+//~^ ERROR encountered dangling pointer
```

Comment:
Maybe add a comment explaining why this is the error that we get?

Presumably, `let x = Wrap(Cell::new(42).as_ptr());` would also dangle in regular code, right?

Reply:
You don't even need `Wrap` in that case. [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=bec8eea17f363fbea68d27cf210fbc7d) shows with miri/mir-dump that the pointer already dangles once `x` is initialized.

Reply:
Yeah, this is the usual "`as_ptr` of a temporary" issue. It should just be explained that that's what happens.